bot/build.sh

```diff
@@ -220,25 +220,18 @@ BUILD_STEP_ARGS+=("--save" "${TARBALL_TMP_BUILD_STEP_DIR}")
 BUILD_STEP_ARGS+=("--storage" "${STORAGE}")
 
 # add options required to handle NVIDIA support
-if command_exists "nvidia-smi"; then
-    # Accept that this may fail
-    set +e
-    nvidia-smi --version
-    ec=$?
-    set -e
-    if [ ${ec} -eq 0 ]; then
-        echo "Command 'nvidia-smi' found, using available GPU"
-        BUILD_STEP_ARGS+=("--nvidia" "all")
-    else
-        echo "Warning: command 'nvidia-smi' found, but 'nvidia-smi --version' did not run succesfully."
-        echo "This script now assumes this is NOT a GPU node."
-        echo "If, and only if, the current node actually does contain Nvidia GPUs, this should be considered an error."
-        BUILD_STEP_ARGS+=("--nvidia" "install")
-    fi
-else
-    echo "No 'nvidia-smi' found, no available GPU but allowing overriding this check"
+# Accept that this may fail
+set +e
+verify_nvidia-smi
+ec=$?
+if [ ${ec} -eq 0 ]; then
+    BUILD_STEP_ARGS+=("--nvidia" "all")
+elif [ ${ec} -eq 1 ]; then
+    BUILD_STEP_ARGS+=("--nvidia" "install")
+elif [  ${ec} -eq 2  ]; then 
```

Comment:
```suggestion
if verify_nvidia-smi; then
    BUILD_STEP_ARGS+=("--nvidia" "all")
else
    # We don't have a GPU but we still want to do a GPU installation
```
Why bother handling the different modes if what you want to do is the same?